wsdl/ver10/schema/onvif.xsd

```diff
@@ -7716,6 +7734,36 @@ and sample rate. </xs:documentation>
 		<xs:anyAttribute processContents="lax"/>
 	</xs:complexType>
 	<!--===============================-->
+	<xs:complexType name="SegmentEncryption">
+		<xs:sequence>
+			<!--When <literal>KeyRotationDuration</literal> is set then the device shall generate a new <literal>KID/Key</literal> pair at the specified time interval as defined by <literal>KeyRotationDuration</literal>, but still use the current key until the segment is finished.-->
+			<xs:any namespace="##any" processContents="lax" minOccurs="0" maxOccurs="unbounded"/>   <!-- first ONVIF then Vendor -->
+		</xs:sequence>
+		<xs:attribute name="CertificateID" type="xs:token" use="required">
+			<xs:annotation>
```

Comment:
When uploading the segment to the external object storage, the device shall include the following attributes in the metadata: i.e CertThumb, Key, KID, IV, EncAlgo. The listed metadata attributes are necessary in order to be able to decrypt the encoded segment. What is the CertificateID here?How this CertificateID used by the client to decrypt the encrypted segment using the listed meat data attributes. what is significance of KID here?

Member
@[venki5685](https://github.com/venki5685) venki5685 [3 days ago](https://github.com/onvif/specs/pull/400#discussion_r1518739625)
Is object metadata also encrypted along with the segment?

Reply:
The CertificateID indicates which certificate's public key is used to encrypt the symmetric key that is added to the object store's metadata.
The full mp4 file is encrypted, while the object's store metadata is not encrypted.
The KID can be used by clients to optimize the decryption process. If the client already has a specific KID, it can use it to decrypt the segment. KID could be the same across segments depending on the value of the key rotation.

Reply:
@jflevesque-genetec, thanks for the clarification.